src/activities/assignments/AssignmentEntity.js

```diff
@@ -379,13 +385,17 @@ export class AssignmentEntity extends Entity {
 	 * @returns {Array} Set of submission type options for this assignment
 	 */
 	submissionTypeOptions() {
-		if (!this.canEditSubmissionType()) {
+		if (!this._entity) {
 			return [];
 		}
 
 		const action = this._entity.getActionByName(Actions.assignments.updateSubmissionType);
+		if (!action) {
+			return this._getReadOnlySubmissionTypeOptions();
+		}
+
 		if (!action.hasFieldByName('submissionType')) {
```

Comment:
Is it possible for this action to exist without that field? Or is this just a safety check.

Reply:
Looks like this is the only place its used, and it has the submission type field (but this is also a safety check) https://search.d2l.dev/xref/lms/le/dropbox/D2L.LE.Dropbox.Extensibility/HM/Folders/Serializers/Siren/SirenFoldersSerializer.cs?r=37b0acf5#1040